x-pack/plugins/ml/public/application/components/custom_urls/custom_url_editor/custom_time_range_picker.tsx

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { FC, useMemo, useState } from 'react';
+import moment, { type Moment } from 'moment';
+import {
+  EuiDatePicker,
+  EuiDatePickerRange,
+  EuiFlexItem,
+  EuiFlexGroup,
+  EuiFormRow,
+  EuiIconTip,
+  EuiSpacer,
+  EuiSwitch,
+} from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { i18n } from '@kbn/i18n';
+import { useMlKibana } from '../../../contexts/kibana';
+
+interface CustomUrlTimeRangePickerProps {
+  onCustomTimeRangeChange: (customTimeRange?: { start: Moment; end: Moment }) => void;
+  customTimeRange?: { start: Moment; end: Moment };
+}
+
+/*
+ * React component for the form for adding a custom time range.
+ */
+export const CustomTimeRangePicker: FC<CustomUrlTimeRangePickerProps> = ({
+  onCustomTimeRangeChange,
+  customTimeRange,
+}) => {
+  const [showCustomTimeRangeSelector, setShowCustomTimeRangeSelectorked1] =
+    useState<boolean>(false);
+  const {
+    services: {
+      data: {
+        query: {
+          timefilter: { timefilter },
+        },
+      },
+    },
+  } = useMlKibana();
+
+  const onCustomTimeRangeSwitchChange = (e: {
```

Comment:
The type of e can be `e: React.ChangeEvent<HTMLInputElement>`

But a neater way might be to just pass in a boolean and call `onCustomTimeRangeSwitchChange` with `onChange={(e) => onCustomTimeRangeSwitchChange(e.target.checked)}`
This is the style we've used the most in our code.

Reply:
Good call - updated to use the consistent style in eca357ab60ac4e8d248f0fa54906aab332d5c42e